syntax = "proto3";
package auth.v1;

service AuthService {
  rpc Register(RegisterRequest) returns (AuthResponse);
  rpc Login(LoginRequest) returns (AuthResponse);
  rpc Logout(LogoutRequest) returns (EmptyResponse);
  rpc RefreshToken(RefreshTokenRequest) returns (AuthResponse);
  rpc ValidateToken(ValidateTokenRequest) returns (ValidateTokenResponse);
  rpc GetCurrentUser(GetCurrentUserRequest) returns (UserResponse);
  rpc UpdateProfile(UpdateProfileRequest) returns (UserResponse);
  rpc DeleteAccount(DeleteAccountRequest) returns (EmptyResponse);
}

message RegisterRequest {
  string email = 1;
  string username = 2;
  string password = 3;
}

message LoginRequest {
  string email = 1;
  string password = 2;
}

message LogoutRequest {
  string refresh_token = 1;
}

message RefreshTokenRequest {
  string refresh_token = 1;
}

message ValidateTokenRequest {
  string access_token = 1;
}

message ValidateTokenResponse {
  bool valid = 1;
  int64 user_id = 2;
  string role = 3;
}

message GetCurrentUserRequest {
  int64 user_id = 1;
}

message UpdateProfileRequest {
  int64 user_id = 1;
  string display_name = 2;
  string bio = 3;
  string avatar_url = 4;
}

message DeleteAccountRequest {
  int64 user_id = 1;
}

message User {
  int64 user_id = 1;
  string email = 2;
  string username = 3;
  string display_name = 4;
  string avatar_url = 5;
  string bio = 6;
  string role = 7;
  bool is_active = 8;
  string created_at = 9;
}

message AuthResponse {
  string access_token = 1;
  string refresh_token = 2;
  string token_type = 3;
  User user = 4;
}

message UserResponse {
  User user = 1;
}

message EmptyResponse {}
